lib/companies_house/client.rb

```diff
@@ -49,6 +49,14 @@ def officers(id)
       items
     end
 
+    def company_search(query, limit)
```

Comment:
It might also be nice to think about using a keyword argument for this, since this gem is [only compatible with Ruby versions which support that](https://github.com/gocardless/companies-house-rest/blob/master/companies-house-rest.gemspec#L23).